### PR TITLE
refactor: unify the greptime artifacts building

### PR DESCRIPTION
**Pull request by zyy17**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)

  Unify the greptime artifacts building and make Dockerfile cleaner. The new Release CI will be refactored base on the new Makefile.

- How does this PR work? Need a brief introduction for the changed logic (optional)

  1. **Use [Docker Buildx](https://docs.docker.com/engine/reference/commandline/buildx/) to build all the images and use one Dockerfile to build images of multiple platforms.**

     - Build docker image of ubuntu base image

       ```
       make docker-image
       ```

     - Build docker image of centos base image

       ```
       make docker-image BASE_IMAGE=centos
       ```

     - Use buildx to build multiple platform images

       It can be very slow, and we don't use it in the Release pipeline instead of building images on the native platform.

       ```
       # Build images of linux/amd64 and linux/arm64.
       make docker-image BUILDX_MULTI_PLATFORM_BUILD=true
       ```

     - **Delete** some old ugly aarch64 Dockerfile

       It's not useful because we all use the one Dockerfile of the same base image.

     However, to speed up the building process in Release CI, we use `make build-greptime-by-buildx` to build `greptime` on the native platforms(amd64 and arm64). After the building is finished, the Release CI will use `ci/Dockerfile*` to build images by copying the binary from the docker builder.

  2. **Add dev-builder image**

     To improve the greptime development experience, I decide to add `docker/dev-builder/Dockerfile` to unify the compiling environment：

     - Build dev-builder image

       ```
       make dev-builder
       ```

     - Use dev-builder to build `greptime`(it'll be slow on macOS because of its virtualization)

       ```
       make build USE_DEV_BUILDER=true
       ```

     - Run integration tests in dev-builder

       ```
       make run-it-in-container
       ```

     For now, the `dev-builder` image is already in DockerHub:

     ```
     docker pull greptime/dev-builder:latest
     ```

  3. **Use `Makefile` as the single entry point for building artifacts** 

     For better maintainability, we try not to expose the raw cargo command(if you're familiar with it, you can still use it). All the Dockerfile and CI(maybe in the next Big PR) should use the `make` command to build the artifacts or run the tests.

     For the above reason, I add `CARGO_PROFILE` / `FEATURES` / `TARGET_DIR` makefile variables to make it more configurable.

  4. **Other improvements**

     - Add `start-etcd` and `stop-etcd` target(use docker to start etcd service, and it will be useful in integration tests);
     - Add `BUILD_JOBS` makefile variable to configure cargo-nextest build jobs;

- Describe clearly one logical change and avoid lazy messages (optional)

- Describe any limitations of the current code (optional)

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

### CODE REVIEW

**Review by sunng87 — APPROVED** [review]

**Comment by v0y4g3r** [comment]
Can I just believe what you committed?

**Review by evenyag — APPROVED** [review]
~~LGTM~~

**Comment by waynexia** [comment]
🛠️ 🚀

**Comment by sunng87** [comment]
*debug after merge*

File: docker/ubuntu/Dockerfile [hunk 1]
```diff
@@ -11,31 +14,42 @@ RUN apt-get update && apt-get install -y \
...
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get \
     -y install ca-certificates \
+    python3.10 \
+    python3.10-dev \
+    python3-pip \
     curl
 
+COPY ./docker/python/requirements.txt /etc/greptime/requirements.txt
+
+RUN python3 -m pip install -r /etc/greptime/requirements.txt
```
Comment (tisonkun) [inline docker/ubuntu/Dockerfile:49]: I'm curious why do we need to install pip requirements in Ubuntu but not in CentOS?

cc @zyy17

**Pull request merged**